src/zenml/zen_stores/schemas/model_schemas.py

```diff
@@ -133,3 +149,248 @@ def update(
+        back_populates="model_version",
+        sa_relationship_kwargs={"cascade": "delete"},
+    )
+
+    version: str = Field(sa_column=Column(TEXT, nullable=False))
+    description: str = Field(sa_column=Column(TEXT, nullable=True))
+    stage: str = Field(sa_column=Column(TEXT, nullable=True))
+
+    @classmethod
+    def from_request(
+        cls, model_version_request: ModelVersionRequestModel
+    ) -> "ModelVersionSchema":
+        """Convert an `ModelVersionRequestModel` to an `ModelVersionSchema`.
+
+        Args:
+            model_version_request: The request model version to convert.
+
+        Returns:
+            The converted schema.
+        """
+        return cls(
+            workspace_id=model_version_request.workspace,
+            user_id=model_version_request.user,
+            model_id=model_version_request.model,
+            version=model_version_request.version,
+            description=model_version_request.description,
+            stage=model_version_request.stage,
+        )
+
+    def to_model(self) -> ModelVersionResponseModel:
+        """Convert an `ModelVersionSchema` to an `ModelVersionResponseModel`.
+
+        Returns:
+            The created `ModelVersionResponseModel`.
+        """
+        return ModelVersionResponseModel(
+            id=self.id,
+            user=self.user.to_model() if self.user else None,
+            workspace=self.workspace.to_model(),
+            created=self.created,
+            updated=self.updated,
+            model=self.model.to_model(),
+            version=self.version,
+            description=self.description,
+            stage=self.stage,
+            model_objects={
+                al.name: al.artifact_id
+                for al in self.objects_links
+                if al.artifact_id is not None and al.is_model_object
+            },
+            deployments={
+                al.name: al.artifact_id
+                for al in self.objects_links
+                if al.artifact_id is not None and al.is_deployment
+            },
+            artifact_objects={
+                al.name: al.artifact_id
+                for al in self.objects_links
+                if al.artifact_id is not None
+                and not (al.is_deployment or al.is_model_object)
+            },
+            pipeline_runs=[
+                al.artifact_id
+                for al in self.objects_links
+                if al.pipeline_run_id is not None
+            ],
+        )
+
+    def update(
+        self,
+        target_stage: str,
+    ) -> "ModelVersionSchema":
+        """Updates a `ModelVersionSchema` to a target stage.
+
+        Args:
+            target_stage: The stage to be updated.
+
+        Returns:
+            The updated `ModelVersionSchema`.
+        """
+        self.stage = target_stage
+        self.updated = datetime.utcnow()
+        return self
+
+
+class ModelVersionLinkSchema(NamedSchema, table=True):
```

Comment:
Somehow this table feels off to me. Not yet fully sure why though.

Some clarifying questions:
- Can a model version be linked to multiple pipeline runs?
- Can an artifact be linked to multiple model versions?
- Are `is_model_object` and `is_deployment` ever `True` for a model-pipeline_run link?

Reply:
- Yes, as many as you wish. Each pipeline run communicating with model - read or update will be tracked here. We define ModelContext for a step or a pipeline, so one pipeline can use many models.
- Good question actually... I thought yes, but we don't plan for logic to support that, cause a context is set at least on a step level. So it might be worth keeping this link table only for Pipeline Runs and putting Artifacts in lists directly in the ModelVersion. @stefannica , @schustmi , WDYT?
- No, always False for those

Reply:
I would upvote to keep M:M for Artifacts in the model, even if we don't use it right away. Maybe spin some tickets for further iterations around Model WatchTower.

Reply:
Ok, let's proceed like this then.